src/common/util/jsonPatcher.js

```diff
@@ -269,12 +357,25 @@ define([
 
         //ovr
         if (sourceJson.ovr && targetJson.ovr) {
-            patch = patch.concat(diff(sourceJson.ovr, targetJson.ovr, '/ovr/', [], true));
-            for (key in targetJson.ovr) {
-                if (sourceJson.ovr[key]) {
-                    patch = patch.concat(
-                        diff(sourceJson.ovr[key], targetJson.ovr[key], '/ovr/' + _strEncode(key) + '/', [], false, key)
-                    );
+            if (sourceJson.ovr.sharded !== true && targetJson.ovr.sharded === true) {
+                // Transformation into sharded overlays means that we have to collect update information.
+                patch.push(getShardingDiff(sourceJson.ovr, targetJson.ovr));
+            } else if (sourceJson.ovr.sharded === true && targetJson.ovr.sharded === true) {
+                patch = patch.concat(diff(sourceJson.ovr, targetJson.ovr, '/ovr/', [], false, '', true, false));
```

Comment:
no.
That case is when the '/ovr' of the coreNode contains only hashes with relids. In the regular case, we deal with the updates of a source entry later (to get the detailed relation changes). In the sharded case, we do not have any more detail (at least not in the regular coreNode), so the updates should be generated as well at that point.